personalrobotics/ada_feeding
# Revamp MoveToMouth/MoveFromMouth with MoveIt2Plan/Execute and BlackboardBehavior

# Description

Since MoveToMouth/MoveFromMouth was originally written, a lot of intermediate classes have been added to the codebase that make it easier to write cleaner code (e.g., #102 #114 #115 ). This PR focuses on moving the old movement trees and its associated behaviors to the new paradigm.

# Testing procedure


Run the code as documented in the README. Then, run the below sequence of actions and verify that they work as expected.

    - [x] `ros2 action send_goal /MoveToMouth ada_feeding_msgs/action/MoveTo "{}" --feedback`
    - [x] `ros2 action send_goal /MoveFromMouthToRestingPosition ada_feeding_msgs/action/MoveTo "{}" --feedback`
    - [x] `ros2 action send_goal /MoveToMouth ada_feeding_msgs/action/MoveTo "{}" --feedback`
    - [x] `ros2 action send_goal /MoveFromMouthToAbovePlate ada_feeding_msgs/action/MoveTo "{}" --feedback`
    - [x] `ros2 action send_goal /MoveToMouth ada_feeding_msgs/action/MoveTo "{}" --feedback`
    - [x] `ros2 action send_goal /MoveFromMouthToRestingPosition ada_feeding_msgs/action/MoveTo "{}" --feedback`
    - [x] `ros2 action send_goal /MoveToMouth ada_feeding_msgs/action/MoveTo "{}" --feedback`
    - [x] `ros2 action send_goal /MoveFromMouthToAbovePlate ada_feeding_msgs/action/MoveTo "{}" --feedback`

# Before opening a pull request

# Before Merging
